### ISSUE TITLE
Issue with attention mask

### ISSUE THREAD

**Issue by xslittlegrass** [0xallam/Direct-Preference-Optimization#1]
In the implementation, the attention mask is setup such that all but the prompt tokens are masked out
```
prompt_prefered_mask = torch.cat([torch.ones_like(prompt_ids), torch.zeros_like(prefered_ids)], dim=-1)
```
This is however inconsistent with the causal attention masked used in pre-training.

I'm wondering whether we should use the same mask so the model behavior will be consistent.

**Comment by ernestchu** [comment]
The author also disregarded the attention_mask comes from `batch_encode_plus`:
https://github.com/0xallam/Direct-Preference-Optimization/blob/8fada75a52dadee252419ec88cee3f25a06ad4a6/src/train.py#L48-L50

**Comment by 0xallam** [comment]
It should be fixed now by #2
